community/kernel/src/main/java/org/neo4j/kernel/api/security/AuthToken.java

```diff
@@ -48,6 +48,19 @@ static String safeCast( String key, Map<String,Object> authToken ) throws Invali
         return (String) value;
     }
 
+    static Map<String,Object> safeCastMap( String key, Map<String,Object> authToken )
+            throws InvalidAuthTokenException
+    {
+        Object value = authToken.get( key );
+        if ( value != null && !(value instanceof Map) )
+        {
+            throw new InvalidAuthTokenException(
+                    "The value associated with the key `" + key + "` must be a Map but was: " +
+                    value.getClass().getSimpleName() );
+        }
+        return (Map<String,Object>) value;
```

Comment:
There's a null-check in the if-case above, but that seems to be there only to prevent NPE from `.getClass()`. This method may still return `null`. Is that intentional?